src/thing/build.test.ts

```diff
@@ -0,0 +1,217 @@
+    nonBuilderThing = adders.addStringWithLocale(
+      nonBuilderThing,
+      "https://some.vocab/predicate",
+      "Yet another string",
+      "en-gb"
+    );
+    nonBuilderThing = removers.removeStringWithLocale(
+      nonBuilderThing,
+      "https://some.vocab/predicate",
+      "Yet another string",
+      "en-gb"
+    );
+    nonBuilderThing = adders.addBoolean(
+      nonBuilderThing,
+      "https://some-other.vocab/predicate",
+      true
+    );
+    nonBuilderThing = setters.setBoolean(
+      nonBuilderThing,
+      "https://some-other.vocab/predicate",
+      false
+    );
+    nonBuilderThing = removers.removeAll(
+      nonBuilderThing,
+      "https://some-other.vocab/predicate"
+    );
+
+    expect(builtThing).toStrictEqual(nonBuilderThing);
+  });
+
+  it("initialises a new Thing if not passed one", () => {
+    const thing = buildThing().build();
+
+    expect(isThing(thing)).toBe(true);
+  });
+
+  it("can take options for the Thing initialisation", () => {
+    const thing = buildThing({
+      url: "https://some.pod/resource#thing",
+    }).build();
+    expect(asIri(thing)).toBe("https://some.pod/resource#thing");
+  });
+
+  it("preserves the type (ThingLocal or ThingPersisted) of Things passed to it", () => {
+    // We're only going to check for expected TypeScript errors:
+    expect.assertions(0);
+
+    // Since we're passing the `url` option, `buildThing` should return a ThingPersisted,
+    // which should cause an error when assigning to a ThingLocal variable:
+    // @ts-expect-error
+    const _thingPersistedDirect: ThingLocal = buildThing({
+      url: "https://some.pod/resource#thing",
+    }).build();
+
+    // Since we're passing a ThingPersisted as the starting Thing,
+    // `buildThing` should also return a ThingPersisted,
+    // which should cause an error when assigning to a ThingLocal variable:
+    // @ts-expect-error
+    const _thingPersisted: ThingLocal = buildThing(
+      createThing({ url: "https://some.pod/resource#thing" })
+    ).build();
+
+    // Since we're passing the `name` option, `buildThing` should return a ThingLocal,
+    // which should not cause an error when assigning to a ThingLocal variable:
+    const _thingLocalDirect: ThingLocal = buildThing({ name: "thing" }).build();
+
+    // Since we're passing a ThingLocal as the starting Thing,
+    // `buildThing` should also return a ThingLocal,
+    // which should not cause an error when assigning to a ThingLocal variable:
+    const _thingLocal: ThingLocal = buildThing(
+      createThing({ name: "thing" })
+    ).build();
+  });
+
+  it("has equivalents for every adder", () => {
+    const adderNames = Object.keys(adders).filter(
+      (adderName) => adderName.substring(0, 3) === "add"
+    );
+    expect.assertions(adderNames.length * 2);
+
+    const builder = buildThing();
+
+    adderNames.forEach((adderName) => {
+      expect(
+        (builder as Record<string, Function>)[adderName]
+      ).not.toBeUndefined();
```

Comment:
Maybe there's a subtle difference, but is equivalent (and more readable)?
```suggestion
      ).toBeDefined();
```

Reply:
Heh, that was mostly my mental model while writing it it seems, leading to me missing the equivalence. bc943d77